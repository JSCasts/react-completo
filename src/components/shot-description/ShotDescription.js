import React from 'react';
import { Link } from 'react-router';

import InsertHtml from '../insert-html/InsertHtml';
import styles from './styles.scss';

const ShotDescription = ({ id, showDescription, description, title }) => {
  if (showDescription) {
    return (
      <Link to={`/shots/${id}`}>
        <div className={styles.container}>
          <h4>{title}</h4>
          <InsertHtml className={styles.description} maxLength={230} text={description} />
        </div>
      </Link>
    );
  }
  return null;
};

ShotDescription.defaultProps = {
  description: '',
};

ShotDescription.propTypes = {
  id: React.PropTypes.number.isRequired,
  showDescription: React.PropTypes.bool.isRequired,
  title: React.PropTypes.string.isRequired,
  description: React.PropTypes.string,
};

export default ShotDescription;
