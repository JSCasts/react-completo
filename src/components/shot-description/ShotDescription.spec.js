import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import ShotDescription from './ShotDescription';

const props = {
  id: 1,
  title: 'title',
  description: 'description',
  showDescription: true,
};

test('Snapshot', () => {
  let component = renderer.create(
    <ShotDescription {...props} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  component = renderer.create(
    <ShotDescription {...props} showDescription={false} />
  );
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

describe('ShotDescription', () => {
  test('has texts', () => {
    let component = shallow(
      <ShotDescription {...props} />
    );
    expect(component.text()).toContain('Link');
    expect(component.html()).toContain(props.title);
    expect(component.html()).toContain(props.description);

    component = shallow(
      <ShotDescription {...props} showDescription={false} />
    );
    expect(component.text()).toContain('');
  });
});
