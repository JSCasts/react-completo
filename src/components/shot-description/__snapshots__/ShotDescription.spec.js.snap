exports[`test Snapshot 1`] = `
<a
  onClick={[Function]}
  style={Object {}}>
  <div
    className="container">
    <h4>
      title
    </h4>
    <div
      className="description">
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "description",
          }
        } />
    </div>
  </div>
</a>
`;

exports[`test Snapshot 2`] = `null`;
